packages/rabbitmq/test/TxSubmitWorker.test.ts

```diff
@@ -212,14 +212,23 @@ describe('TxSubmitWorker', () => {
 
     const promises: Promise<void>[] = [];
     const result = [undefined, undefined, undefined, undefined, undefined, undefined, undefined];
+    let res: unknown = null;
 
     for (let i = 0; i < 7; ++i) {
       promises.push(rabbitMqTxSubmitProvider.submitTx(txs[i].txBodyUint8Array));
       // Wait 10ms to be sure the transactions are enqueued in the right order
       await new Promise((resolve) => setTimeout(resolve, 10));
     }
 
-    await expect(Promise.all(promises)).resolves.toEqual(result);
+    try {
+      res = await Promise.all(promises);
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (error: any) {
+      expect(error.innerError).toBeUndefined();
```

Comment:
Sorry @rhyslbw ,

here the expected behavior is to not throw; the problem was that in case of error, jest logs only the outer error which usually is "unknown error inspect innerError".

So here the logic is: if line 217 throws the test (correctly) fails due to line 221, but if `error.innerError` is not `undefined`, the test will fail for line 220 (rather than 221) which will let jest to log the `innerError`.

So, in case we would need to add `expect.assertions(1);` as the expected behavior is that only assertion at line 224 will be executed, but in this case should not be strictly required as if the flow enters the catch `block`, the test will fail anyway.

Do you agree?

Reply:
> here the expected behavior is to not throw; the problem was that in case of error

I'm finding it hard to reason about a test that has assertions we're not expecting to be called. What's causing the inconsistency here?

Reply:
Before the change, we had only line 224:
```typescript
res = await Promise.all(promises);
```
and it is expected to not throw...

Sometimes it happens (reported by @IvayloAndonov and maybe @mkazlauskas too) that this test fails due to an `Error` thrown by that line.

Jest behavior in case of unhandled exception is to log only `Error.message`, which too often is something like "unknown error inspect innerError"

The purpose of this change is to increase the verbosity in such cases.
Now, in case of `Error` thrown by `Promise.all`, the first check performed is
```typescript
expect(error.innerError).toBeUndefined()
```
which, in such cases, will fail in a way which let jest to log the details of `error.innerError`.

In any other error case (i.e. when there is no `innerError`), the first check will pass but not the next one
```typescript
expect(error).toBeUndefined()
```
It will fail and jest will log the details of `error`.

P.S.
The [logger module](https://github.com/input-output-hk/cardano-js-sdk/pull/366#discussion_r933349543) I spoken about should also log `innerError` by itself when required so we could avoid tricks like this one. 😉

Reply:
Ah right, this is to assist with debugging the intermittent test failures. You can ignore this request then, thanks for the explanation!

Reply:
Exactly! That's the intent of this change.